launch.py

```diff
@@ -1,48 +1,61 @@
-from modules import launch_utils
-
-args = launch_utils.args
-python = launch_utils.python
-git = launch_utils.git
-index_url = launch_utils.index_url
-dir_repos = launch_utils.dir_repos
-
-commit_hash = launch_utils.commit_hash
-git_tag = launch_utils.git_tag
-
-run = launch_utils.run
-is_installed = launch_utils.is_installed
-repo_dir = launch_utils.repo_dir
-
-run_pip = launch_utils.run_pip
-check_run_python = launch_utils.check_run_python
-git_clone = launch_utils.git_clone
-git_pull_recursive = launch_utils.git_pull_recursive
-list_extensions = launch_utils.list_extensions
-run_extension_installer = launch_utils.run_extension_installer
-prepare_environment = launch_utils.prepare_environment
-configure_for_tests = launch_utils.configure_for_tests
-start = launch_utils.start
-
-
-def main():
-    if args.dump_sysinfo:
-        filename = launch_utils.dump_sysinfo()
-
-        print(f"Sysinfo saved as {filename}. Exiting...")
-
-        exit(0)
-
-    launch_utils.startup_timer.record("initial startup")
-
-    with launch_utils.startup_timer.subcategory("prepare environment"):
-        if not args.skip_prepare_environment:
-            prepare_environment()
-
-    if args.test_server:
-        configure_for_tests()
-
-    start()
-
-
-if __name__ == "__main__":
-    main()
+import socket
+from modules import launch_utils
+
+args = launch_utils.args
+python = launch_utils.python
+git = launch_utils.git
+index_url = launch_utils.index_url
+dir_repos = launch_utils.dir_repos
+
+commit_hash = launch_utils.commit_hash
+git_tag = launch_utils.git_tag
+
+run = launch_utils.run
+is_installed = launch_utils.is_installed
+repo_dir = launch_utils.repo_dir
+
+run_pip = launch_utils.run_pip
+check_run_python = launch_utils.check_run_python
+git_clone = launch_utils.git_clone
+git_pull_recursive = launch_utils.git_pull_recursive
+list_extensions = launch_utils.list_extensions
+run_extension_installer = launch_utils.run_extension_installer
+prepare_environment = launch_utils.prepare_environment
+configure_for_tests = launch_utils.configure_for_tests
+start = launch_utils.start
+
+
+def check_network_connectivity():
+    try:
+        socket.create_connection(("www.github.com", 80))
+        return True
+    except OSError:
+        return False
+
+
+def main():
+    if args.dump_sysinfo:
+        filename = launch_utils.dump_sysinfo()
+
+        print(f"Sysinfo saved as {filename}. Exiting...")
+
+        exit(0)
+
+    launch_utils.startup_timer.record("initial startup")
+
+    with launch_utils.startup_timer.subcategory("prepare environment"):
+        if not args.skip_prepare_environment:
+            if check_network_connectivity():
```

Comment:
this will not do

your change will prevent web UI from launching if they are disconnected from the internet or not able to reach GitHub for whatever reason
webui should be able to launch without a connection after it's installed and assets are downloaded

webui should not make any unnecessary connections to any external servers including GitHub
if you wish to add better error messages about cloning repos, go ahead
but don't test if connectivity is available

you can already see get already fails the only issue is that most people don't understand the error messages